pkg/config/node.go

```diff
@@ -437,7 +443,7 @@ func GetConfig(kubeconfigPath, clusterConfigPath, resolvConfPath string, apiVips
 		} else {
 			ingressVip = nil
 		}
-		newNode, err := getNodeConfig(kubeconfigPath, clusterConfigPath, resolvConfPath, apiVip, ingressVip, apiPort, lbPort, statPort)
+		newNode, err := getNodeConfig(kubeconfigPath, clusterConfigPath, resolvConfPath, apiVip, ingressVip, apiPort, lbPort, statPort, apiLBIPs[0], apiIntLBIPs[0])
```

Comment:
Adding unit tests. To answer the other question, apiLBIPs[0], apiIntLBIPs[0] would be "" by default. It will have actual values only when the `userProvisionedDNS` feature is enabled.

Reply:
Sorry if I miss something obvious but I don't see that... You call `GetConfig` by passing `[]net.IP{}` as `apiLBIPs` and then you reference `apiLBIPs[0]`. This does not work, try running that code

```
a := []net.IP{}
fmt.Println(a[0])
```

and the ourput is not `""` but a panic `runtime error: index out of range [0] with length 0`

Reply:
Thanks @mkowalski ! I needed to update the logic in that part of the code too.

Reply:
I would really recommend starting from unit tests and then proceed with the implementation... What I can see in the current code is that it is still panicking because `[]net.IP{}` and `nil` are different entities (i.e. empty slice and `nil` is not the same in golang).

Look at the following code

```
a := []net.IP{}
if a == nil {
	fmt.Println("nil")
} else {
	fmt.Println(a[0])
}
```

or run it via playground, e.g. https://go.dev/play/p/UbrxBrUstTw